extensions/src/test/java/org/pentaho/test/platform/web/http/security/RequestParameterAuthenticationFilterTest.java

```diff
@@ -97,4 +109,125 @@ public void userNamePasswordUnencrypted() throws IOException, ServletException {
     verify( authManagerMock ).authenticate( Mockito.eq( authRequest ) );
 
   }
+
+
+  @Test
+  public void testSetCsrfProcessorRespectsIt() {
+
+    CsrfProcessor csrfProcessorMock = mock( CsrfProcessor.class );
+    RequestParameterAuthenticationFilter filter = new RequestParameterAuthenticationFilter();
+
+    // ---
+
+    filter.setCsrfProcessor( csrfProcessorMock );
+
+    // ---
+
+    assertSame( csrfProcessorMock, filter.getCsrfProcessor() );
+  }
+
+  private void testRequestParameterAuthenticationFailsThenThrowsCsrfValidationAuthException( @NonNull Throwable validateError )
+          throws ServletException, IOException {
+
+    CsrfProcessor csrfProcessorMock = mock( CsrfProcessor.class );
+
+    final MockHttpServletRequest request =
+            new MockHttpServletRequest( "GET",
+                    "http://localhost:9080/pentaho-di/kettle/executeTrans/?rep=dev&userid=admin&password=password&trans=/home"
+                            + "/admin/Trans1" );
+    request.addParameter( "userid", "admin" );
+    request.addParameter( "password", "password" );
```

Comment:
Given that `addParameter` is explicitly called in the mock, is there a real need for the URL to have the same query string parameters?